grails-app/controllers/au/org/emii/netcdfsubset/SubsetController.groovy

```diff
@@ -21,13 +23,29 @@ class SubsetController {
         response.setContentType("application/octet-stream")
         response.setHeader("Content-disposition", "filename=${filenameToServe(subsetRequest)}")
 
-        try {
-            subsetService.subset(subsetRequest.typeName, subsetRequest.CQL_FILTER, response.outputStream)
+        if (isThreadAvailable()) {
+            try {
+                getThreadLock()
+                subsetService.subset(subsetRequest.typeName, subsetRequest.CQL_FILTER, response.outputStream)
+            }
+            catch (InterruptedException e) {
+                log.info "The maximum number of concurrent jobs has been exceeded", e
+                render status: 503, text: "The maximum number of concurrent jobs has been exceeded, please try again later"
+            }
+            catch (Exception e) {
+                log.error "Unhandled exception during subset", e
+                releaseThreadLock()
```

Comment:
Is it necessary to release in this catch handler, if it's released in the finally block?